go/vt/vttablet/tabletmanager/vreplication/utils.go

```diff
@@ -82,46 +91,53 @@ func getLastLog(dbClient *vdbClient, vreplID int32) (id int64, typ, state, messa
 	return id, typ, state, message, nil
 }
 
-func insertLog(dbClient *vdbClient, typ string, vreplID int32, state, message string) error {
+func insertLog(dbClient *vdbClient, typ string, vreplID int32, state, message string) {
 	// getLastLog returns the last log for a stream. During insertion, if the type/state/message match we do not insert
 	// a new log but increment the count. This prevents spamming of the log table in case the same message is logged continuously.
 	id, _, lastLogState, lastLogMessage, err := getLastLog(dbClient, vreplID)
 	if err != nil {
-		return err
+		log.Errorf("Could not insert vreplication_log record because we failed to get the last log record: %v", err)
+		return
 	}
 	if typ == LogStateChange && state == lastLogState {
 		// handles case where current state is Running, controller restarts after an error and initializes the state Running
-		return nil
+		return
 	}
 	var query string
 	if id > 0 && message == lastLogMessage {
 		query = fmt.Sprintf("update %s.vreplication_log set count = count + 1 where id = %d", sidecar.GetIdentifier(), id)
 	} else {
 		buf := sqlparser.NewTrackedBuffer(nil)
+		// We perform the truncation, if needed, in the middle of the message as the end of the message is likely to
```

Comment:
You should just extract this logic into a  helper, which just does the fitting to size. That will also simplify the unit test ...
Also the for loop probably can be replaced to make it O(1). 

```
const prefixFactor = 0.4

func FitToSize(message string, maxLen int, truncationIndicator string) string {
	l := len(message)
	availableLen := maxLen - len(truncationIndicator)

	if l <= availableLen {
		return message
	}

	prefixLen := int(float64(availableLen) * prefixFactor)
	suffixLen := availableLen - prefixLen

	return message[:prefixLen] + truncationIndicator + message[l-suffixLen:]
}
```